service-netty/src/main/java/info/bitrich/xchangestream/service/netty/NettyStreamingService.java

```diff
@@ -179,7 +179,7 @@ protected void initChannel(SocketChannel ch) {
                                 }
 
                                 handlers.add(handler);
-                                p.addLast(handlers.toArray(new ChannelHandler[handlers.size()]));
+                                p.addLast(handlers.toArray(new ChannelHandler[0]));
```

Comment:
Why this change?  The existing code was fractionally more efficient than this (good discussion on it here: https://stackoverflow.com/questions/3293946/the-easiest-way-to-transform-collection-to-array)

Reply:
JetBrains Idea said that better to use empty array in ArrayTo function. :)

Reply:
reference
intellij
Call to 'toArray()' with pre-sized array argument 'new PathMatcher[collection.size()]' less... (Ctrl+F1)
There are two styles to convert a collection to an array: either using a pre-sized array (like c.toArray(new String[c.size()])) or using an empty array (like c.toArray(new String[0]).
In older Java versions using pre-sized array was recommended, as the reflection call which is necessary to create an array of proper size was quite slow.
However since late updates of OpenJDK 6 this call was intrinsified, making the performance of the empty array version the same and sometimes even better, compared to the pre-sized version.
Also passing pre-sized array is dangerous for a concurrent or synchronized collection as a data race is possible between the size and toArray call which may result in extra nulls at the end of the array,
if the collection was concurrently shrunk during the operation.
This inspection allows to follow the uniform style: either using an empty array (which is recommended in modern Java) or using a pre-sized array (which might be faster in older Java versions or non-HotSpot based JVMs).

Reply:
https://medium.com/@hanru.yeh/collection-toarray-new-t-0-will-be-faster-than-collection-toarray-new-t-size-5b129fabfa88

Reply:
I am both convinced and educated :)